pkg/registry/registrytest/etcd.go

```diff
@@ -149,6 +150,27 @@ func (t *Tester) TestWatch(valid runtime.Object, labelsPass, labelsFail []labels
 }
 
 // =============================================================================
+// get codec based on runtime.Object
+func getCodec(obj runtime.Object) (runtime.Codec, error) {
+	_, kind, err := api.Scheme.ObjectVersionAndKind(obj)
+	if err != nil {
+		return nil, fmt.Errorf("unexpected encoding error: %v", err)
+	}
+	// TODO: caesarxuchao: we should detect which group an object belongs to
+	// by using the version returned by Schem.ObjectVersionAndKind() once we
+	// split the schemes for internal objects.
+	// TODO: caesarxuchao: we should add a map from kind to group in Scheme.
+	var codec runtime.Codec
+	if api.Scheme.Recognizes(testapi.Default.GroupAndVersion(), kind) {
```

Comment:
If we end up with kinds that are registered in multiple groups, then we'll need to actually store the group in every object. But I don't think we're going to have that case. At any rate, soon we'll make api.Scheme keep a map so it will just know what group an object is from.